diff --git a/nethack-3.4.3/src/hacklib.c b/nethack-3.4.3/src/hacklib.c
index 0d08270..904356e 100644
--- a/nethack-3.4.3/src/hacklib.c
+++ b/nethack-3.4.3/src/hacklib.c
@@ -457,12 +457,14 @@ extern struct tm *FDECL(localtime,(time_t *));
 #endif
 static struct tm *NDECL(getlt);
 
+
 void
 setrandom()
 {
 	/* the types are different enough here that sweeping the different
 	 * routine names into one via #defines is even more confusing
 	 */
+#if 0 //no random
 #ifdef RANDOM	/* srandom() from sys/share/random.c */
 	srandom((unsigned int) time((time_t *)0));
 #else
@@ -483,6 +485,7 @@ setrandom()
 #  endif
 # endif
 #endif
+#endif //no random
 }
 
 static struct tm *
